src/main/java/org/sonarqube/gradle/SonarPropertyComputer.java

```diff
@@ -143,12 +163,74 @@ private void computeSonarProperties(Project project, Map<String, Object> propert
     for (Project childProject : enabledChildProjects) {
       String moduleId = childProject.getPath();
       moduleIds.add(moduleId);
-      String modulePrefix = (prefix.length() > 0) ? (prefix + "." + moduleId) : moduleId;
-      computeSonarProperties(childProject, properties, modulePrefix);
+      String modulePrefix = !prefix.isEmpty() ? (prefix + "." + moduleId) : moduleId;
+      computeDefaultProperties(childProject, properties, modulePrefix);
     }
+
     properties.put(convertKey("sonar.modules", prefix), String.join(",", moduleIds));
   }
 
+  private boolean shouldApplyScanAll(Project project, Map<String, Object> properties) {
+    // when the parent module is skipped, the properties are empty thus the scan all logic is not applied
+    var scanAllValue = (String) properties.getOrDefault(SONAR_GRADLE_SCAN_ALL, "false");
+    var scanAllEnabled = "true".equalsIgnoreCase(scanAllValue.trim());
+
+    if (scanAllEnabled) {
+      LOGGER.info("Parameter sonar.gradle.scanAll is enabled. The scanner will attempt to collect additional sources.");
+
+      // Collecting the properties configured in the Gradle build configuration
+      var sonarProps = new SonarProperties(new HashMap<>());
+      actionBroadcastMap.get(project.getPath()).execute(sonarProps);
+
+      var sourcesOrTestsAlreadySet = Stream.of(System.getProperties().keySet(), Utils.loadEnvironmentProperties(System.getenv()).keySet(), sonarProps.getProperties().keySet())
+        .flatMap(Collection::stream)
+        .anyMatch(k -> ScanProperties.PROJECT_SOURCE_DIRS.equals(k) || ScanProperties.PROJECT_TEST_DIRS.equals(k));
+
+      if (sourcesOrTestsAlreadySet) {
+        LOGGER.warn("Parameter sonar.gradle.scanAll is enabled but the scanner will not collect additional sources because sonar.sources or sonar.tests are already set.");
+        return false;
+      }
+    }
+
+    return scanAllEnabled;
+  }
+
+  private static void computeScanAllProperties(Project project, Map<String, Object> properties) {
+    Set<Path> existingSources = SonarUtils.splitAsCsv((String) properties.get(ScanProperties.PROJECT_SOURCE_DIRS))
+      .stream()
+      .filter(Predicate.not(String::isBlank))
+      .map(Paths::get)
+      .collect(Collectors.toSet());
+
+    // TODO: clarify the desired logic: when the skipProject property is set on a module should we skip the whole module and all its submodules?
+    Set<Path> skippedDirs = project.getAllprojects()
+      .stream()
+      .filter(SonarQubePlugin::isSkipped)
+      .map(Project::getProjectDir)
+      .map(File::toPath)
+      .collect(Collectors.toSet());
+
+    SourceCollector visitor = new SourceCollector(existingSources, skippedDirs, Set.of(), false);
+
+    try {
+      Files.walkFileTree(project.getProjectDir().toPath(), visitor);
+    } catch (IOException e) {
+      LOGGER.warn(String.valueOf(e));
```

Comment:
Question: Is an early return safer here?

Reply:
We have this behavior on the scanner for Maven as well. We should change it on both projects to fail